src/GCPClientManager.php

```diff
@@ -6,17 +6,71 @@
 
 use Google\Cloud\ResourceManager\V3\FoldersClient;
 use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Google\Cloud\ResourceManager\V3\ProjectsClient;
+use Google\Cloud\ServiceUsage\V1\ServiceUsageClient;
+use Google_Client;
+use Google_Service_CloudResourceManager;
+use Google_Service_Iam;
 
 class GCPClientManager
 {
+    public const SCOPES_CLOUD_PLATFORM = 'https://www.googleapis.com/auth/cloud-platform';
+
+    /** @var array<FoldersClient|ProjectsClient|ServiceUsageClient> */
+    private array $clients = [];
+
     /**
      * @throws \Google\ApiCore\ValidationException
      */
     public function getFoldersClient(GenericBackendCredentials $credentials): FoldersClient
     {
-        return new FoldersClient([
+        $client = new FoldersClient([
+            'credentials' => $this->getCredentialsArray($credentials),
+        ]);
+        $this->clients[] = $client;
+
+        return $client;
+    }
+
+    public function getProjectClient(GenericBackendCredentials $credentials): ProjectsClient
+    {
+        $client = new ProjectsClient([
+            'credentials' => $this->getCredentialsArray($credentials),
+        ]);
+
+        $this->clients[] = $client;
+
+        return $client;
+    }
+
+    public function getServiceUsageClient(GenericBackendCredentials $credentials): ServiceUsageClient
+    {
+        $client = new ServiceUsageClient([
+            'credentials' => $this->getCredentialsArray($credentials),
+        ]);
+
+        $this->clients[] = $client;
```

Comment:
Nie kazdy klient ma tu `->close()` metodu

Reply:
dal bych buď k těm klientom co nemajů close do tych metod, že nemá close tak ho nenastavuješ do this->clients, nebo udělal method_exists close v tom destruktoru takto podle kódu nepoznáš proč něco ukládáš a něco ne.

Reply:
napisal som to k tym ktore nemaju close ako komentar